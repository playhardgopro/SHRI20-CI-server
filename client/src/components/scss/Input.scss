@import './var.scss';
.input {
  border: 2px solid var(--color-bg-border);
  border-radius: 4px;
  &:focus {
    border: 2px solid var(--color-bg-border-focused);
  }
}
.input:focus ~ .input__controls-append .button::after {
  border-color: var(--color-bg-border-focused);
}

.input ~ .input__controls-append .icon {
  color: var(--blue-grey-250);
}

.input ~ .input__controls-append .button:hover .icon {
  color: var(--color-bg-border-focused);
}

.input_size_m {
  height: var(--text-line-height-36);
  padding-right: var(--space-13);
  padding-left: var(--space-13);
}

.input_width_full {
  flex: 1 1 auto;
}
.input_width_52 {
  width: 52px;
  max-width: 100px;
}

.input__group {
  display: flex;
  align-items: center;
}

.input__group_vertical {
  align-items: stretch;
  flex-direction: column;
}

@media screen and (max-width: $screen-m) {
  .input__group_vertical.input__group_m-horizontal .input__label {
    margin-bottom: 0;
  }
}
.input__group_vertical .input__label {
  margin-right: 0;
  margin-bottom: var(--space-6);
}

.input__label {
  margin-right: var(--space-8);
}

.input__label_required::after {
  color: var(--red-600);
  content: ' *';
}

.input__controls {
  display: flex;
}

.input__controls-append {
  display: flex;
  align-items: center;
  margin-left: -2px;
}
.input__controls-append .text {
  margin-left: calc(var(--space-8) + 2px);
}

.input__controls-append .button {
  height: 100%;
  margin-left: -24px;
  background-color: var(--color-bg-main);
}
.input__controls-append .button:after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  content: '';
  border: 2px solid var(--color-bg-border);
  border-left: 0;
  border-top-left-radius: 0;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 0;
}
.input:focus .input__controls-append .button::after {
  border: 2px solid var(--color-bg-border-focused);
}

.input__group > .input:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.input:not(:placeholder-shown) + .input__controls-append .button {
  display: none;
}
