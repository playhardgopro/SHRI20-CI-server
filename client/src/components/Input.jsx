import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import { Text, Icon, Button } from '.'
import './scss/Input.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const cnInput = cn('input')

const textStyle = { size: '13-18', type: 'h2' }

const LilInput = ({ id, placeholder, isShort }) => {
  const inputClass = { size: 'm', width: isShort ? 52 : 'full' }
  return (
    <input className={cnInput(inputClass, ['text', { size: '13-15' }])} id={id} type="text" placeholder={placeholder} />
  )
}

const ControlsAppend = ({ text }) => {
  return text ? (
    <Text className={{ size: '13-18' }}>minutes</Text>
  ) : (
    <button type="button" className="button button_size_m button_distribute_center button_view_control">
      <Icon name="close" className={{ size: 'm' }} />
    </button>
  )
}

const Input = ({ children, className, options }) => {
  const { placeholder, label, isRequired, id, vertical, text } = options
  return (
    <div className={cnInput('group', { vertical })}>
      <label className={cnInput('label', { required: isRequired })} htmlFor="repository">
        <Text className={textStyle}>{label}</Text>
      </label>
      <div className={cnInput('controls')}>
        <LilInput id={id} placeholder={placeholder} isShort={text} />
        <div className={cnInput('controls-append')}>{ControlsAppend({ text })}</div>
      </div>
    </div>
  )
}

Input.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Input.defaultProps = {
  children: '',
  className: {},
  options: { label: '', placeholder: '' },
}

export default Input
