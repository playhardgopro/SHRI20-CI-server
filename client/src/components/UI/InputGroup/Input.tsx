import * as React from 'react'
import { withNaming } from '@bem-react/classname'
import { Text, Button } from '../../index'
import './InputGroup.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })
const cnInput = cn('input')
const cnText = cn('text')

type name = 'repoName' | 'buildCommand' | 'mainBranch' | 'period'
interface InputProps {
  clearable?: boolean
  id: name
  name: name
  width?: 'full' | 52
  size?: 's' | 'm'
  placeholder: string
  onClear(name: name): void
  text?: string
  status: 'invalid' | 'valid' | undefined
  className?: {}
  inputRef: (ref: HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement | null) => void
}

export const Input: React.FC<InputProps> = ({
  id,
  name,
  clearable,
  onClear,
  inputRef,
  placeholder,
  status,
  className,
  text,
  width,
  size,
}) => {
  const handleClear = () => {
    onClear(name)
  }
  return (
    <div className={cnInput('controls')}>
      <input
        id={id}
        name={name}
        ref={inputRef}
        placeholder={placeholder}
        className={cnInput({ ...className, status, width, size }, [cnText({ size: '13-15' })])}
      />
      <div className={cnInput('controls-append')}>
        {clearable && (
          <Button
            className={{ size: 'm', distribute: 'center', view: 'control' }}
            icon={{ name: 'close', size: 'm' }}
            onClick={handleClear}
            empty
          />
        )}
        {text && <Text className={{ size: '13-18' }}>{text}</Text>}
      </div>
    </div>
  )
}

export default Input
