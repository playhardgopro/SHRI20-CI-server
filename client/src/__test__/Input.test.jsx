import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../components/UI/InputGroup/Input'

describe('Input', () => {
  it('renders correctly', () => {
    const input = renderer.create(<Input id="test" name="test" value="1234" clearable />).toJSON()

    expect(input).toMatchSnapshot()
  })
  it('empty renders correctly', () => {
    const input = renderer.create(<Input id="test" name="test" placeholder="test" clearable />).toJSON()

    expect(input).toMatchSnapshot()
  })
  it('short renders correctly', () => {
    const input = renderer.create(<Input id="test" name="test" value="1234" width="52" text="minutes" />).toJSON()

    expect(input).toMatchSnapshot()
  })
  it('invalid renders correctly', () => {
    const input = renderer.create(<Input id="test" name="test" value="1234" status="invalid" clearable />).toJSON()

    expect(input).toMatchSnapshot()
  })
})
