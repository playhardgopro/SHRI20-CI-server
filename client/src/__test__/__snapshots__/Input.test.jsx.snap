// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input empty renders correctly 1`] = `
<div
  className="input__controls"
>
  <input
    className="input input_width_full input_size_m text text_size_13-15"
    id="test"
    name="test"
    placeholder="test"
  />
  <div
    className="input__controls-append"
  >
    <button
      className="button button_size_m button_distribute_center button_view_control"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon_size_m"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16C3.6 16 0 12.4 0 8C0 3.6 3.6 0 8 0C12.4 0 16 3.6 16 8C16 12.4 12.4 16 8 16ZM12 5.12L10.88 4L8 6.88L5.12 4L4 5.12L6.88 8L4 10.88L5.12 12L8 9.12L10.88 12L12 10.88L9.12 8L12 5.12Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`Input invalid renders correctly 1`] = `
<div
  className="input__controls"
>
  <input
    className="input input_status_invalid input_width_full input_size_m text text_size_13-15"
    id="test"
    name="test"
    placeholder=""
  />
  <div
    className="input__controls-append"
  >
    <button
      className="button button_size_m button_distribute_center button_view_control"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon_size_m"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16C3.6 16 0 12.4 0 8C0 3.6 3.6 0 8 0C12.4 0 16 3.6 16 8C16 12.4 12.4 16 8 16ZM12 5.12L10.88 4L8 6.88L5.12 4L4 5.12L6.88 8L4 10.88L5.12 12L8 9.12L10.88 12L12 10.88L9.12 8L12 5.12Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`Input renders correctly 1`] = `
<div
  className="input__controls"
>
  <input
    className="input input_width_full input_size_m text text_size_13-15"
    id="test"
    name="test"
    placeholder=""
  />
  <div
    className="input__controls-append"
  >
    <button
      className="button button_size_m button_distribute_center button_view_control"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon_size_m"
      >
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16C3.6 16 0 12.4 0 8C0 3.6 3.6 0 8 0C12.4 0 16 3.6 16 8C16 12.4 12.4 16 8 16ZM12 5.12L10.88 4L8 6.88L5.12 4L4 5.12L6.88 8L4 10.88L5.12 12L8 9.12L10.88 12L12 10.88L9.12 8L12 5.12Z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`Input short renders correctly 1`] = `
<div
  className="input__controls"
>
  <input
    className="input input_width_52 input_size_m text text_size_13-15"
    id="test"
    name="test"
    placeholder=""
  />
  <div
    className="input__controls-append"
  >
    <span
      className="text text_size_13-18"
    >
      minutes
    </span>
  </div>
</div>
`;
